### ISSUE TITLE
[lec4] activation function은 여기서 설명한 것들 말고도 더 있나요? activation function들을 왜 activation function이라고 하나요?

### ISSUE THREAD

**Issue by positive235** [ai-robotics-kr/CS231n_study#25]
## activation function은 여기서 설명한 것 말고도 더 있나요?

> softmax function도 activation function이라고 구글링하니까 나오네요.

## activiation function들을 왜 activation function이라고 하나요?

> non-linear eqation이라서 linear가 non-linear에 들어가서 complexity를 높일 수 있어서 activation function이라고 알고 있는데요. activation function 중에 쓸수 있는게 미분이 가능한 것들만 쓸수 있는걸로 알고 있어요. step function같은거는 activation function으로 안 쓰이는 이유이기도 하고요

> 찾아봤는데 어느 블로그에서 "activation function decides whether a neuron should be activated or not by calculated weighted sum and further adding bias with it."라는데요, 원래 보통은 wx + b 이렇게 하는데 sigmoid할 때는 바로 sigmoid(wx+b)를 취하지 sigmoid구할 때 또 가중치를 곱하고 바이어스를 더하진 않잖아요 그렇게 하지 않고 activate 시킬 수 있는거를 activation function이라고 하는 것 같고, activation function의 목적은 비선형을 뉴런의 아웃풋 값에 도입하는것? 영어로는 
The purpose of activation function is to introduce non-linearity into the output of the neuron. 이렇게 geeksforgeeks에 설명이 나와있네요.

**Comment by positive235** [comment]
카카오톡 의견: 

[Activation Function 관련]

비선형성 추가를 위해 레이어 중간에 사용시 sigmoid : vanishing gradient problem : 이를 대체하는 것이 ReLU지만 이도 사실 만능은 아니고 음수 부분이 비활성화되므로 dying ReLU문제가 존재.  LeakyReLU같은 다양한 변종들 : GAN과 같은 네트워크에서 주로 사용
http://dataaspirant.com/2017/03/07/difference-between-softmax-function-and-sigmoid-function/
https://stats.stackexchange.com/a/287294/246401

**Comment by positive235** [comment]
카카오톡 의견:

### Softmax function
- Used for multi-classification in logistic regression model.
- the probabilities sum will be 1
- Used in the different layers of neural networks
- The high value will have the higher probability than other values
### Sigmoid function
- Used for binary classification in logistic regression model
- the probabilities sum need not be 1
- used as activation function while building neural network
- the high value will have the high probability but not the higher probability

**Comment by positive235** [comment]
슬랙 의견:

좋은 정보가 있어서 공유해요 :slightly_smiling_face:
지금까지 렉처에서는 sigmoid(wx+b)를 쌓고 쌓아서 뉴럴네트워크를 만든다고 배웠는데, 그게 머릿속에선 잘 그려지지 않는 편이죠? ㅎㅎ
colah 블로그라고 딥러닝에 대해 굉장히 설명 잘 하는 블로그가 있는데, 여기서 sigmoid(wx+b)를 단순히 숫자로 하는 linear algebra + non-linear operation이 아니라 공간의 변환과 휘어짐으로 classification하는 방법을 보여주네요! 이걸 Manifolds라는 개념으로 보여줍니다 :slightly_smiling_face:
https://colah.github.io/posts/2014-03-NN-Manifolds-Topology/

**Comment by positive235** [comment]
정혜원님 슬랙 의견:
cross entropy loss function을 derive하는데 softmax가 사용되는거같습니다
코드랑 수식 보시면 될거같습니다
https://deepnotes.io/softmax-crossentropy

**Comment by positive235** [comment]
장형기 님 슬랙 의견:

보편적으로 classification probabilities를 구할 때 softmax를 쓰는데, 굳이 exp()를 걸지 않고 그냥 vector sum으로 나누면 안되나?라고 생각했었습니다
는데 stack exchange에서 같은 질문을 한 사람이 있더라구요 ㅎㅎ
그냥 vector sum으로 나누는 경우에는, vector sum = 0인 경우에 singularity가 생겨서 안된다고 하네요!
https://stats.stackexchange.com/questions/419751/why-is-softmax-function-used-to-calculate-probabilities-although-we-can-divide-e
